src/main/java/ladysnake/spawnercontrol/SpawnerEventHandler.java

```diff
@@ -211,6 +215,54 @@ public static void onLivingExperienceDrop(LivingExperienceDropEvent event) {
         }
     }
 
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public static void onLivingItemDrop(LivingDropsEvent event) {
+        EntityLivingBase livingBase = event.getEntityLiving();
+        SpawnerConfig cfg = SpawnerUtil.getConfig(livingBase.getEntityData().getTag(NBT_TAG_SPAWNER_POS));
+        if (cfg != null) {
+            try {
+                ResourceLocation rl = EntityList.getKey(event.getEntity());
+                if (rl != null) {
+                    SpawnerConfig.MobLoot.MobLootEntry entry = cfg.mobLoot.lootEntries.get(rl);
+                    List<EntityItem> drops = event.getDrops();
+                    if (entry.removeAllItems) {
+                        drops.clear();
+                    }
+                    else {
+                        for (EntityItem drop : drops.toArray(new EntityItem[0])) {
+                            ItemStack stack = drop.getItem();
+                            for (String s : entry.removedItems) {
+                                String[] split = s.split(":");
+                                if (stack.getUnlocalizedName().equals(split[0] + ":" + split[1])
+                                        && (split.length < 3 || stack.getMetadata() == Integer.parseInt(split[2]))) {
+                                    drops.remove(drop);
+                                }
+                            }
+                        }
+                    }
+                    World world = livingBase.world;
+                    double x = livingBase.posX, y = livingBase.posY, z = livingBase.posZ;
+                    for (String s : entry.addedItems)
+                    {
+                        String[] split = s.split(":");
+                        if (split.length < 5 || world.rand.nextFloat() < Double.parseDouble(split[4])) {
+                            ResourceLocation itemRL = new ResourceLocation(split[0], split[1]);
+                            Item item = ForgeRegistries.ITEMS.getValue(itemRL);
+                            if (item == null)
+                            {
+                                //Try/catch handles this if null
+                                item = Item.getItemFromBlock(ForgeRegistries.BLOCKS.getValue(itemRL));
+                            }
+                            drops.add(new EntityItem(world, x, y, z, new ItemStack(item, split.length < 3 ? 1 : Integer.parseInt(split[2]), split.length < 4 ? 0 : Integer.parseInt(split[3]))));
```

Comment:
The values computed from `split` should each be stored in a variable with an explicit name. This line is too long, and the parameters to `new ItemStack()` are not immediately obvious.

Reply:
I opted not to put the "chance" (index 4) part of the split strings into an explicit variable, because it would either prevent a possible short-circuit optimization (which is not a big deal in this case, but still bugs me), or add another level of bracing (and an additional boolean)

If you want the chance in an explicit var anyway, just poke me one more time and I'll do it lol

Leaving just this one "unresolved" in case you still want changes to it